### ISSUE TITLE
[junit-platform-engine] Extra events emmited during run

### ISSUE THREAD

**Issue by ocherniavskyi-lohika-tix** [cucumber/cucumber-jvm#2233]
**Describe the bug**
handleFeatureStartedHandler emmited for every feature despite cucumber expression was provided

**To Reproduce**
1. Create at least two feature files with one scenario each
2. Mark one of the tests with unique tag
3. Start test run with given tag expression
4. Check how many times handleFeatureStartedHandler event emmited

**Expected behavior**
handleFeatureStartedHandler should be emmited only for those features that have scenarios that suites provided tag expression

**Actual behavior**
handleFeatureStartedHandler emmited for every feature files, even if no scenario will be executed due to tag expression

**Context & Motivation**
This bug makes impossible to use some plugin, e.g. https://github.com/allure-framework/allure-java/tree/master/allure-cucumber6-jvm

**Your Environment**
- cucumber core 6.9.1
- cucumber-junit-platform-engine 6.9.1
 - junit-platform-console-standalone 1.7.1

**Comment by ocherniavskyi-lohika-tix** [comment]
@mpkorstanje hi, could you check this please?

**Comment by mpkorstanje** [comment]
You've left it rather vague how this make it impossible to use a plugin. Perhaps you can clarify with a concrete example.

I guess you are confusing the method name `handleFeatureStartedHandler` from the Allure plugin with the `TestSourceRead` event from Cucumber. The purpose of the `TestSourceRead` event is to provide plugins with the source of the feature that has been read. If no scenarios from that feature file were executed, a plugin can choose not include it in its output.

**Comment by mpkorstanje** [comment]
Feel free to reopen this with more information.

**Comment by ocherniavskyi-lohika-tix** [comment]
The problem is TestSourceRead triggered only once when starting tests on Junit4 runner or cucumber console runner, so i dont think plugins should handle such ambiguous behavior

**Comment by ocherniavskyi-lohika-tix** [comment]
@mpkorstanje should i create new issue?

**Comment by mpkorstanje** [comment]
The test source read event is emitted for each feature read and before that feature is executed.

Now you seem to expect something but you aren't making it clear what it is that you expect. Rather then describing it, could you demonstrate the difference? For example by making a plugin that prints out the events it received and showing the difference.

**Comment by ocherniavskyi-lohika-tix** [comment]
@mpkorstanje i cant reopen this bug
